runtime/interpreter/decode.go

```diff
@@ -1161,3 +1164,24 @@ func (d *Decoder) decodeRestrictedStaticType(v interface{}) (StaticType, error)
 		Restrictions: restrictions,
 	}, nil
 }
+
+func (d *Decoder) decodeType(v interface{}) (TypeValue, error) {
+	encoded, ok := v.(map[interface{}]interface{})
+	if !ok {
+		return TypeValue{}, fmt.Errorf("invalid type encoding")
+	}
+
+	decodedStaticType, err := d.decodeStaticType(encoded[uint64(0)])
```

Comment:
why using `0`? because [we defined here](https://github.com/onflow/cadence/pull/184/files#diff-b4059fff0bb0bd33abbff410a57cec00R1077)?

Reply:
Correct. I really don't like that, but so far I haven't found a better way

Reply:
Maybe we could define a const right above where it was used for reference:

```
const (
  EncodedTypeValueType uint64 = 0
)

type encodedTypeValue struct {
	Type interface{} `cbor:"0,keyasint"`
}
```

And then here, we can do 
```
	decodedStaticType, err := d.decodeStaticType(encoded[EncodedTypeValueType])

```

Reply:
I opened #216 to improve this for all keys